cmd/eib/main.go

```diff
@@ -66,6 +71,37 @@ func processArgs() (*image.Context, error) {
 		KubernetesArtefactDownloader: kubernetes.ArtefactDownloader{},
 	}
 
+	failedValidations := validation.ValidateDefinition(ctx)
+	if len(failedValidations) > 0 {
+		audit.Audit("Image definition validation found the following errors:")
+
+		failuresByComponent := map[string][]validation.FailedValidation{}
+		logMessageBuilder := strings.Builder{}
+
+		for _, fv := range failedValidations {
+			failuresByComponent[fv.Component] = append(failuresByComponent[fv.Component], fv)
```

Comment:
The repetitive nature of this makes me believe that `ValidateDefinition()` returning a `map[string][]validation.FailedValidation` is the way to go here. We'd be dropping the component initialisation for each struct and we'd be completely dropping the loop here in favour of the one below.

Reply:
Yeah, I agree. This sort of translation like this is a very Python thing to do so I hadn't even stepped back to consider that `ValidateDefinition` should return them by component.